slave/buildslave/scripts/start.py

```diff
@@ -111,8 +111,22 @@ def startSlave(basedir, quiet, nodaemon):
 
     # we probably can't do this os.fork under windows
     from twisted.python.runtime import platformType
+
     if platformType == "win32":
-        return launch(nodaemon)
+        pidfile = os.path.join(basedir, 'twistd.pid')
+        # Make sure the twistd.pid file actually exists
+        if not os.path.exists(pidfile):
+            with open(pidfile, "w") as f:
+                f.write("{}".format(os.getpid()))
+
+        launch(nodaemon)
+
+        # When Buildbot is closing (which is now), delete the PID
```

Comment:
The reason I prefer it being here is because it better supports if the buildslave process ends by something other than calling Buildslave Stop (such as an ordinary stop signal).